pkg/api/user.go

```diff
@@ -33,7 +33,7 @@ func getUserUserProfile(userId int64) Response {
 
 // GET /api/users/loginoremail/:loginOrEmail
 func GetUserByLoginOrEmail(c *middleware.Context) Response {
-	query := m.GetUserByLoginQuery{LoginOrEmail: c.Params(":loginOrEmail")}
+	query := m.GetUserByLoginQuery{LoginOrEmail: c.QueryEscape("loginOrEmail")}
```

Comment:
For simple cases it returns correct results, however I agree that it is not the right one to be used here. On the other hand macaron doesn't provide a simple func for url string queries. The only option AFAICS which is close to what we are trying to achieve is the Query() which takes also into account post requests for forms. Use directly the `c.Req.URL.Query().Get("loginOrEmail")` might be a good and simple option though.

I have two more concerns regarding the lookup and email query that would like to discuss a bit further.
1) How do you handle the RFC compliance for emails on such cases? For example if a user provide an email address which contains special characters in local-part.
2) Security-wise it is a bit of a bad practice to pass personal information (in this case the email address) via URL with a GET request. Perhaps a more suitable solution would be to use POST instead and pass the parameter via the body. However this doesn't follow very well the REST guidelines.